ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -197,15 +207,30 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         auto& eps = _ip_data[ip].eps;
         auto const& sigma_eff = _ip_data[ip].sigma_eff;
 
-        double const S = _process_data.specific_storage(t, x_position)[0];
         double const K_over_mu =
             _process_data.intrinsic_permeability(t, x_position)[0] /
             _process_data.fluid_viscosity(t, x_position)[0];
         auto const alpha = _process_data.biot_coefficient(t, x_position)[0];
+        auto const K_S = solid_material.getBulkModulus(t, x_position);
         auto const rho_sr = _process_data.solid_density(t, x_position)[0];
-        auto const rho_fr = _process_data.fluid_density(t, x_position)[0];
+        // TODO get fluid properties from GPML
+        double const rho_fr =
+            (_process_data.fluid_type == FluidType::INCOMPRESSIBLE_FLUID)
+                ? _process_data.fluid_density(t, x_position)[0]
+                : (_process_data.fluid_type == FluidType::COMPRESSIBLE_FLUID)
+                      ? _process_data.fluid_density(t, x_position)[0]
+                      : (_process_data.fluid_type == FluidType::IDEAL_GAS)
+                            ? N_p.dot(p) / (R_s * T_ref)
+                            : 0;
```

Comment:
I think this could be written more readable using a [lambda expression](https://en.cppreference.com/w/cpp/language/lambda).